river/Output.zig

```diff
@@ -185,227 +163,259 @@ pub fn sendViewTags(self: Self) void {
+    const visible_count_str = std.fmt.allocPrint(self.root.server.allocator, "{}", .{visible_count}) catch |err| switch (err) {
+        error.OutOfMemory => {
+            @panic("out of memory");
+        },
+    };
+    const master_count_str = std.fmt.allocPrint(self.root.server.allocator, "{}", .{self.master_count}) catch |err| switch (err) {
+        error.OutOfMemory => {
+            @panic("out of memory");
+        },
+    };
+    const master_factor_str = std.fmt.allocPrint(self.root.server.allocator, "{d}", .{self.master_factor}) catch |err| switch (err) {
+        error.OutOfMemory => {
+            @panic("out of memory");
+        },
+    };
+    const layout_width_str = std.fmt.allocPrint(self.root.server.allocator, "{}", .{layout_width}) catch |err| switch (err) {
+        error.OutOfMemory => {
+            @panic("out of memory");
+        },
+    };
+    const layout_height_str = std.fmt.allocPrint(self.root.server.allocator, "{}", .{layout_height}) catch |err| switch (err) {
+        error.OutOfMemory => {
+            @panic("out of memory");
+        },
+    };
 
-    var i: u32 = 0;
-    var it = ViewStack(View).pendingIterator(self.views.first, output_tags);
-    while (it.next()) |node| {
-        const view = &node.view;
+    // Create command to be executed. All commands are executed via sh and the
+    // five layout parameters are appended to the end.
+    const layout_command = std.mem.join(self.root.server.allocator, " ", &[_][]const u8{
+        self.layout,
+        visible_count_str,
+        master_count_str,
+        master_factor_str,
+        layout_width_str,
+        layout_height_str,
+    }) catch |err| {
+        Log.Error.log("Error assemblind layout command.", .{});
+        return false;
+    };
+    const cmd = [_][]const u8{
+        "/bin/sh",
+        "-c",
+        layout_command,
+    };
 
-        if (view.floating) {
-            continue;
-        }
+    // Execute layout executable
+    const child = std.ChildProcess.init(&cmd, self.root.server.allocator) catch |err| {
+        Log.Error.log("Error initializing layout command.", .{});
+        return false;
+    };
+    child.stdin_behavior = .Ignore;
+    child.stdout_behavior = .Pipe;
+
+    std.ChildProcess.spawn(child) catch |err| {
+        Log.Error.log("Error executing layout command.", .{});
+        child.deinit();
+        self.root.server.allocator.free(layout_command);
+        return false;
+    };
 
-        var new_box: Box = undefined;
-        new_box = .{
-            .x = x_offset,
-            .y = y_offset,
-            .width = layout_width,
-            .height = layout_height,
+    const max_output_size = 400 * 1024;
+    const buffer = child.stdout.?.inStream().readAllAlloc(self.root.server.allocator, max_output_size) catch |err| {
+        child.deinit();
+        self.root.server.allocator.free(layout_command);
+        return false;
+    };
```

Comment:
same as above, add a `defer` to free the memory and remove later individual calls to free.